src/components/pages/WelcomePage.tsx

```diff
@@ -75,35 +75,34 @@ class WelcomePage extends React.Component<Props, State> {
    * Component did mount life-cycle handler
    */
   public componentDidMount = async () => {
-    window.addEventListener("scroll", this.handleScroll);
     this.setState({
       loading: true
     });
-
     const api = ApiUtils.getApi();
-    const [posts, mainMenu, localeMenu, popularCategory, media, customizeFields] = await Promise.all(
-      [
-        api.getWpV2Posts({per_page: 90}),
-        api.getMenusV1LocationsById({ lang: this.props.lang, id: "main" }),
-        api.getMenusV1LocationsById({ lang: this.props.lang, id: "locale" }),
-        api.getWpV2Categories({ slug: ["popular"] }),
-        api.getWpV2Media({}),
-        api.getWpV2Customize()
-      ]
-    );
 
-    const categoryIdArray = [(popularCategory.length > 0 ? popularCategory[0].id || -1 : -1).toString()];
+    const customizeFields = await api.getWpV2Customize();
+
+    this.setState({
+      customizeFields: customizeFields
+    });
+
+    const posts = await api.getWpV2Posts({per_page: 90});
+    const mainMenu = await api.getMenusV1LocationsById({ lang: this.props.lang, id: "main" });
+    const localeMenu = await api.getMenusV1LocationsById({ lang: this.props.lang, id: "locale" });
+    const popularCategory = await api.getWpV2Categories({ slug: ["suosittu"] });
+    const media = await api.getWpV2Media({per_page: 50});
```

Comment:
Why make all calls separate? If you wish to get customize fields first you could still fetch everything else after that in Promise.all(). I think this version now waits between each call.

Reply:
I've pushed new commit, where it's done how suggested